app/services/proofing/resolution/progressive_proofer.rb

```diff
@@ -50,40 +52,144 @@ def proof(
           user_email:,
         )
 
-        instant_verify_residential_address_result = instant_verify_residential_address_plugin.call(
+        residential_address_resolution_result = residential_address_plugin.call(
           applicant_pii:,
           current_sp:,
           ipp_enrollment_in_progress:,
           timer:,
         )
 
-        instant_verify_state_id_address_result = instant_verify_state_id_address_plugin.call(
+        state_id_address_resolution_result = state_id_address_plugin.call(
           applicant_pii:,
           current_sp:,
-          instant_verify_residential_address_result:,
+          residential_address_resolution_result:,
           ipp_enrollment_in_progress:,
           timer:,
         )
 
         state_id_result = aamva_plugin.call(
           applicant_pii:,
           current_sp:,
-          instant_verify_state_id_address_result:,
+          state_id_address_resolution_result:,
           ipp_enrollment_in_progress:,
           timer:,
         )
 
         ResultAdjudicator.new(
           device_profiling_result: device_profiling_result,
           ipp_enrollment_in_progress: ipp_enrollment_in_progress,
-          resolution_result: instant_verify_state_id_address_result,
+          resolution_result: state_id_address_resolution_result,
           should_proof_state_id: aamva_plugin.aamva_supports_state_id_jurisdiction?(applicant_pii),
           state_id_result: state_id_result,
-          residential_resolution_result: instant_verify_residential_address_result,
+          residential_resolution_result: residential_address_resolution_result,
           same_address_as_id: applicant_pii[:same_address_as_id],
           applicant_pii: applicant_pii,
         )
       end
+
+      def proofing_vendor
+        @proofing_vendor ||= begin
+          default_vendor = IdentityConfig.store.idv_resolution_default_vendor
+          alternate_vendor = IdentityConfig.store.idv_resolution_alternate_vendor
+          alternate_vendor_percent = IdentityConfig.store.idv_resolution_alternate_vendor_percent
+
+          # Initially we will have production environments configured with
+          # `proofer_mock_fallback: false`
+          # We need to honor that configuration until all environments are
+          # updated to use idv_resolution_default_vendor
+          if !IdentityConfig.store.proofer_mock_fallback
+            if default_vendor == :mock
+              return :instant_verify
+            end
+          end
+
+          if default_vendor.blank?
+            raise InvalidProofingVendorError, 'idv_resolution_default_vendor not configured'
```

Comment:
issue: This type of configuration loading/validation should happen during startup automatically. Consider moving additional validation to an initializer.

Reply:
This is a good call-out. I'll look into how we handle this in intializers

Reply:
The closest we have to config validation prior art is [unused_identity_config_keys.rb](https://github.com/18F/identity-idp/blob/main/config/initializers/unused_identity_config_keys.rb), but setting up globally available config is probably best demonstrated by the [telephony initializer](https://github.com/18F/identity-idp/blob/main/config/initializers/unused_identity_config_keys.rb).

Reply:
Would using an enum in the definition of the config work?

As an example of an existing one:

```ruby
    config.add(
      :openid_connect_redirect,
      type: :string,
      enum: ['server_side', 'client_side', 'client_side_js'],
    )
```